src/index.ts

```diff
@@ -30,20 +36,59 @@ const defaultLog: EdgeLog = Object.assign(
   }
 )
 
-export async function createRepo(
-  syncKey: string,
-  opts: CreateRepoOptions = {}
-): Promise<void> {
+async function apiRequest<T>(
+  request: ApiRequest,
+  opts: OptionParams = {}
+): Promise<ApiResponse<T>> {
   const { fetch = nodeFetch, log = defaultLog } = opts
+  const { method, url, body } = request
+
+  const start = Date.now()
+  const response = await fetch(url, {
+    method,
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(body)
+  })
+  const timeElapsed = Date.now() - start
+
+  log(`${method} ${url} returned ${response.status} in ${timeElapsed}ms`)
+
+  if (!response.ok)
+    throw new Error(`Request ${method} ${url} failed ${response.status}`)
+
+  const responseData: ApiResponse<T> = await response.json()
```

Comment:
Should this be cleaned?

Reply:
ApiResponse does not have a cleaner. But, I am interested in making it into a cleaner. I am curious as to how we could achieve this.